internals/daemon/api_notices.go

```diff
@@ -182,5 +261,19 @@ func v1GetNotice(c *Command, r *http.Request, _ *UserState) Response {
 	if notice == nil {
 		return statusNotFound("cannot find notice with ID %q", noticeID)
 	}
+	if !noticeViewableByUser(notice, requestUID, daemonUID) {
+		return statusForbidden("not allowed to access notice with id %q", noticeID)
+	}
 	return SyncResponse(notice)
 }
+
+func noticeViewableByUser(notice *state.Notice, requestUID, daemonUID uint32) bool {
```

Comment:
nitpick: it's slightly asymmetric that state has filtering logic but this bit here, maybe part of this should be a predicate method on Notice itself?

Reply:
I don't think we want the state to be aware of whether a user is admin (and certainly shouldn't be aware of `daemonUID`, but I do think we could have a function like this in state:
```go
func (n *Notice) ViewableByUser(userID uint32) {
    return n.userID == nil || userID == *n.userID
}
```

Then the usage in the daemon would be
```go
if isAdmin(requestUID, daemonUID) || notice.ViewableByUser(requestUID)
```

I don't know if this is actually clearer though, since a notice's viewability is very dependent on whether the user is admin, and this requires all users of the current `noticeViewableByUser` to separately check for `isAdmin` as well.

@benhoyt what do you think about this?

Reply:
Since the notices filter only applies to the `state.Notices()` function, not `state.Notice()`, we could add an analogous `userID *uint32` parameter to `state.Notice()`, but I don't really like that, since that again doesn't account for admin rights, and presumably wouldn't differentiate between a notice with that ID not existing and the notice existing but the user ID not matching. So I still think the daemon should be fully responsible for permissions checking, and we consider filtering to be an expedient but not inherently permissions-centric way to implement permission checking in the daemon for `GetNotices()`.

Reply:
> So I still think the daemon should be fully responsible for permissions checking

Yeah, I agree. This suggestion seems like it would be moving (half of) the permission checking into state. I think we should leave any concept of "viewable by" where it is in the daemon.